packages/wrangler/src/pages/download-config.ts

```diff
@@ -67,9 +74,24 @@ async function toEnvironment(
 	const configObj = {} as RawEnvironment;
 	configObj.compatibility_date =
 		project.compatibility_date ?? new Date().toISOString().substring(0, 10);
+
+	// Find the latest supported compatibility date and use that
+	if (project.always_use_latest_compatibility_date) {
```

Comment:
a bit after the fact, but I found the `project` terminology here a bit confusing, given that Pages has the concept of a `Project`, and that, in this particular case, `project` is in fact referring to the deployment config.